lib/modules/datasource/rpm/index.ts

```diff
@@ -0,0 +1,220 @@
+        logger.warn(`Empty response body from getting ${filelistsGzipUrl}.`);
+        throw new Error(
+          `Empty response body from getting ${filelistsGzipUrl}.`,
+        );
+      }
+      // decompress the gzipped file
+      decompressedBuffer = await gunzipAsync(response.body);
+    } catch (err) {
+      logger.warn(
+        `Failed to fetch or decompress ${filelistsGzipUrl}: ${
+          err instanceof Error ? err.message : err
+        }`,
+      );
+      throw err;
+    }
+    const xmlString = decompressedBuffer.toString('utf8');
+    // check if filelistsXmlUrl is in XML format
+    if (!(xmlString.startsWith('<?xml') || xmlString.startsWith('\n<?xml'))) {
+      logger.warn(
+        `Decompressed ${filelistsGzipUrl} is not in XML format. Contents: ${xmlString}`,
+      );
+      throw new Error(
+        `Decompressed ${filelistsGzipUrl} is not in XML format. Contents: ${xmlString}`,
+      );
+    }
+
+    // parse filelists.xml
+    const data = new XmlDocument(xmlString);
+    const packages = data.childrenNamed('package');
+    if (!packages || packages.length === 0) {
+      logger.warn(
+        `No packages found in ${filelistsGzipUrl}, xml contents: ${xmlString}`,
+      );
+      throw new Error(`No packages found in ${filelistsGzipUrl}`);
+    }
+    const releases: Record<string, Release> = {};
+    logger.debug(`Found ${packages.length} packages in ${filelistsGzipUrl}`);
+    for (const pkg of packages) {
+      logger.debug(
+        `Checking package ${pkg.attr.name} for releases in ${filelistsGzipUrl}`,
+      );
+      const name = pkg.attr.name;
+      if (name !== packageName) {
+        continue;
+      }
+      const versionElement = pkg.childNamed('version');
+      const version = versionElement?.attr?.ver ?? '';
+      if (!version) {
+        logger.debug(
+          `No version found for package ${name} in ${filelistsGzipUrl}`,
+        );
+        continue;
+      }
+      const rel = versionElement?.attr?.rel;
+      let versionWithRel = version;
+      if (rel) {
+        // if rel is present, we need to append it to the version. Otherwise, ignore it.
+        // e.g. 1.0.0-1, 1.0.0-2, 1.0.0-3 or 1.0.0
+        logger.debug(
+          `Found version ${version} with rel ${rel} for package ${name}`,
+        );
+        versionWithRel += `-${rel}`;
+      }
+      logger.debug(
+        `Found version ${versionWithRel} for package ${name} in ${filelistsGzipUrl}`,
+      );
+
+      // check if the versionWithRel isn't already in the releases
+      // One version could have multiple rel
+      // (note: this rel is the release/revision key in filelists.xml, not the release data type)
+      // e.g. 1.0.0-1, 1.0.0-2, 1.0.0-3
+      if (!releases[versionWithRel]) {
+        releases[versionWithRel] = {
+          version: versionWithRel,
+        };
+      }
+    }
+    // if no releases were found for the package, return null
+    if (Object.keys(releases).length === 0) {
+      logger.debug(
```

Comment:
```suggestion
      logger.trace(
```

will be logged by upper layer too

Reply:
updated